packages/react-ui-components/src/TextInput/textInput.js

```diff
@@ -94,21 +120,37 @@ class TextInput extends PureComponent {
             return <div key={key}>{validationError}</div>;
         });
 
+        const inputRef = el => {
+            this.inputRef = el;
+        };
+
         return (
             <div className={containerClassName}>
                 <input
                     {...rest}
                     className={classNames}
                     role="textbox"
+                    type={type}
                     placeholder={placeholder}
                     disabled={disabled}
                     onChange={this.handleValueChange}
+                    onKeyPress={this.handleKeyPress}
+                    ref={inputRef}
                     />
                 {renderedErrors && <TooltipComponent>{renderedErrors}</TooltipComponent>}
             </div>
         );
     }
 
+    handleKeyPress(e) {
+        const enterKeyCode = 13;
+        const keyCode = e.keyCode || e.which;
+        const {onEnterKey} = this.props;
+        if (keyCode === enterKeyCode) {
+            onEnterKey();
```

Comment:
Imho, we don't know at this point, whether `onEnterKey` is set. Please add a check before executing it.